Yafc/Workspace/ProductionTable/ProductionTableView.cs

```diff
@@ -131,11 +142,16 @@ public override void BuildElement(ImGui gui, RecipeRow recipe) {
 
                             BuildFavorites(imgui, recipe.recipe, "Add recipe to favorites");
 
-                            if (recipe.subgroup != null && imgui.BuildRedButton("Delete nested table").WithTooltip(imgui, recipe.subgroup.expanded ? "Shortcut: Collapse, then right-click" : "Shortcut: right-click") && imgui.CloseDropdown()) {
+                            if (recipe.subgroup != null && imgui.BuildRedButton("Delete nested table").WithTooltip(
+                                imgui, recipe.subgroup.expanded ? "Shortcut: Collapse, then right-click" :
+                                "Shortcut: right-click") && imgui.CloseDropdown()) {
+
                                 _ = recipe.owner.RecordUndo().recipes.Remove(recipe);
                             }
 
-                            if (recipe.subgroup == null && imgui.BuildRedButton("Delete recipe").WithTooltip(imgui, "Shortcut: right-click") && imgui.CloseDropdown()) {
+                            if (recipe.subgroup == null && imgui.BuildRedButton("Delete recipe").WithTooltip(imgui,
+                                "Shortcut: right-click") && imgui.CloseDropdown()) {
```

Comment:
Wrapped code should be more readable than the original. To make this happen, line breaks should not be added within an expression unless they have also been added on both sides of the expression.

Including the leading whitespace, this line is less than half the width of my screen, so I'm not convinced it needs line breaks at all. But if it does, they should be added in this order:
```c#
if (recipe.subgroup == null
    // (1) Add breaks at operators before adding them within the expressions passed to those operators
    && imgui.BuildRedButton("Delete recipe")
        // (2) Add breaks before .MethodName before adding breaks between method parameters
        .WithTooltip(imgui,
            // (3) Add breaks between method parameters before adding breaks within method parameters
            "Shortcut: right-click")
    // (1)
    && imgui.CloseDropdown()) {
```
I generally prefer to put binary operators at the beginning of a line, but `,` and `=>` go at the end of the line instead. I'm not entirely sure why.

Reply:
> this line is less than half the width of my screen

That's true. The reason for that is the mind-boggling degree of nesting. I haven't looked into how to fix nesting yet, but yes, I'm aware that sometimes first 24 characters are empty, and that's not good.

Reply:
```
if (recipe.subgroup == null
    // (1) Add breaks at operators before adding them within the expressions passed to those operators
    && imgui.BuildRedButton("Delete recipe")
```
Hmm, the operators like `&&` on the next line. My initial thought was to leave them on the same line so the operator indicates that that was not the end of the expression. However, I may be wrong. I'll need to read on the conventions in this regard.

Reply:
On the other hand, there needs to be consistency between the operators. I definitely remember that the dot operator goes on the next line. I'll remake the PR to follow that rule. Thank you for the check!

Reply:
Re: why `,` and `=>` go onto the previous line:

`=>` belongs to what's before it, method arguments => method body. It's common to put a method body on a new line, but also the have that indent one more.

Similar to `,`. If you put the `,` at the end of a line, it's probably a list of values, and the previous line should also just contain that list. Thus you can signal you're adding a multi line item with extra indentation, or a new list item by not adding indentation. If that make sense.

I agree other operators like `.`, `&&`, `||` should be at the start of the next line.

> My initial thought was to leave them on the same line so the operator indicates that it was not the end of the expression. 

The lack of closing brackets or semicolon already does that. As well as the indentation of the next line.

Reply:
To extend on the above: Either do not split method arguments, split all of them, so:

```C#
.WithTooltip(imgui, "Shortcut: right-click")
```
or
```C#
.WithTooltip(
    imgui,
    "Shortcut: right-click")
```

Also if you're calling a fluent api with mulitple calls, the closing bracket can go to the next line:

```C#
.WithTooltip(
    imgui,
    "Shortcut: right-click"
).OtherCalll()
```

Reply:
I haven't found much about lambda-wrapping online. Even [google C# code style](https://google.github.io/styleguide/csharp-style.html) doesn't specify it. 

So I asked in a C# [discord](https://discord.gg/csharp) of 11k people. 
I was told that the arrow on a new line is prevalent in the code they saw before.

Reply:
Interesting. I'm mostly exposed to the Microsoft learn docs, and it's never at the start of the line there as far as i know, almost always in the middle or the end. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions

Reply:
> I generally prefer to put binary operators at the beginning of a line, but `,` and `=>` go at the end of the line instead.

Thinking more about this, it's not `=>` that goes at the end of the line, but  `=> {`. `=>` not followed by `{` goes at the beginning of the line for the same reason `||`, `+`, or `.` do. However, if you apply the "operators at the beginning of the line" rule to `=> {`, you get a line that contains only `=> {`. This violates the spirit of the rule that puts `{` at the end of the line, rather than on its own line.

Reply:
@DaleStan I've addressed your point in https://github.com/shpaass/yafc-ce/pull/270/commits/95ee7cb26ab59d1c6aad1dcbf6dafebdc916da45.
Please resolve this discussion if you consider the argument settled.

Reply:
@DaleStan 
> `=>` not followed by `{` goes at the beginning of the line for the same reason `||`, `+`, or `.` do.

I haven't found the mention of this aspect in the Google C# Style [Guide](https://google.github.io/styleguide/csharp-style.html). 

However, Google Java Style [Guide](https://google.github.io/styleguide/javaguide.html) has the following enty:
> A line is never broken adjacent to the arrow in a lambda, except that a break may come immediately after the arrow if the body of the lambda consists of a single unbraced expression.

```
Predicate<String> predicate = str ->
    longExpressionInvolving(str);
```

What do you think?